src/Client/GuzzleFileFetcher.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace Tuf\Client;
+
+use GuzzleHttp\Client;
+use GuzzleHttp\ClientInterface;
+use GuzzleHttp\Exception\RequestException;
+use Tuf\Exception\DownloadSizeException;
+use Tuf\Exception\RepoFileNotFound;
+use Tuf\JsonNormalizer;
+
+/**
+ * Defines a file fetcher that uses Guzzle to read a file over HTTPS.
+ */
+class GuzzleFileFetcher implements RepoFileFetcherInterface
+{
+    /**
+     * The HTTP client.
+     *
+     * @var \GuzzleHttp\ClientInterface
+     */
+    private $client;
+
+    /**
+     * GuzzleFileFetcher constructor.
+     *
+     * @param \GuzzleHttp\ClientInterface $client
+     *   The HTTP client.
+     */
+    public function __construct(ClientInterface $client)
+    {
+        $this->client = $client;
+    }
+
+    /**
+     * Creates an instance of this class with a specific base URI.
+     *
+     * @param string $baseUri
+     *   The base URI from which to fetch files.
+     *
+     * @return static
+     *   A new instance of this class.
+     */
+    public static function createFromUri(string $baseUri) : self
+    {
+        $scheme = parse_url($baseUri, PHP_URL_SCHEME);
+        if ($scheme === 'https') {
+            $client = new Client(['base_uri' => $baseUri]);
+            return new static($client);
+        } else {
+            throw new \InvalidArgumentException("Repo base URI must be HTTPS: $baseUri");
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function fetchFile(string $fileName, int $maxBytes)
+    {
+        try {
+            $response = $this->client->request('GET', $fileName);
+        } catch (RequestException $e) {
```

Comment:
Could we change this to `\GuzzleHttp\Exception\ClientException` it says it covers "4xx codes"

Reply:
How about \GuzzleHttp\Exception\BadResponseException instead? That'll cover 5xx codes too.

Reply:
but we don't care about 5xx codes. We are just going to rethrow anyways. So seems like we should be as narrow as we can. If they had a 404Except I would say you use that but there isn't one